docs/pages/upgrading/client-tools-managed-updates.mdx

```diff
@@ -0,0 +1,273 @@
+---
+title: Teleport Client Tool Managed Updates
+description: Explains how to use Teleport client tools (`tsh` and `tctl`) managed updates.
+labels:
+ - conceptual
+ - platform-wide
+---
+
+This documentation explains how to keep Teleport client tools like `tsh` and `tctl` up-to-date.
+Updates can be managed by cluster or self-managed, ensuring tools are secure, free from bugs,
+and compatible with your Teleport cluster.
+
+Why keep client tools updated?
+
+- **Security**: Updates deliver patches for known vulnerabilities.
+- **Bug Fixes**: Resolved issues are pushed to endpoints.
+- **Compatibility**: Avoid manual understanding of [Teleport component compatibility rules](overview.mdx#component-compatibility).
+
+## How it works
+
+When you run `tsh login`, the tsh tool will check if updates are enabled for your cluster.
+If your client version differs from the cluster's required version, it will:
+
+1. Download the updated version.
+2. Validate the package with a checksum.
+3. Extract the package and store the binaries in `~/.tsh/bin` (`$TELEPORT_HOME/bin`).
+4. Records the client tools version advertised by the cluster in `~/.tsh/bin/.config.json`(`$TELEPORT_HOME/bin/.config.json`).
+5. Re-execute using the updated version (with the same environment variables).
+
+### Key features
+
+**Binary Management**: Original binaries are preserved, and updates are stored separately.
+Updates are installed in the `$TELEPORT_HOME/bin` folder (if `TELEPORT_HOME` is not defined, the home folder is used `$HOME/.tsh/bin`).
+When client tools (`tctl` or `tsh`) are executed from any other path, they consistently check for binaries in the update
+folder and re-execute them if found.
+
+**Validation**: Downloaded packages are verified with a hash sum to ensure integrity.
+Package downloads are directed to the `cdn.teleport.dev` endpoint and depend on the operating system,
+platform, and edition. The edition must be identified by the original client tools binary.
+The URL pattern is as follows:
+- `https://cdn.teleport.dev/teleport-{ent-}vX.Y.Z-{linux,darwin,windows}-{amd64,arm64,arm,386}-{fips-}bin.{tar.gz,pkg,zip}`
+- `https://cdn.teleport.dev/teleport-{ent-}vX.Y.Z-{linux,darwin,windows}-{amd64,arm64,arm,386}-{fips-}bin.{tar.gz,pkg,zip}.sha256`
+The base URL of the CDN can be overridden using the `TELEPORT_CDN_BASE_URL` environment variable.
+Allows mirroring the CDN in a private network or using custom builds.
+
+**Concurrency**: Tools use a locking mechanism to enable smooth operation during updates.
+Only one process can acquire the lock to update client tools, while other processes wait for the lock to be released.
+If the first process cancels the update, the next process in line will initiate the update.
+
+### Multi-cluster support and multi-version caching
+
+Client tools managed updates support multi-cluster environments. When you log in to different clusters that use
```

Comment:
Can we say "since version X.Y.Z"? Or maybe only in the backport?

Reply:
I going add this to each backport separately